CMakeLists.txt

```diff
@@ -25,6 +25,28 @@ else()
 	set(CVD_INTERNAL_ARCH_LITTLE_ENDIAN ON)
 endif()
 
+option(CVD_SSE2 "Enable SSE2" OFF)
+option(CVD_SSE "Enable SSE" OFF)
+option(CVD_FAST "Enable FAST" ON)
+option(CVD_TESTS "Enable tests" ON)
+option(CVD_PROGS "Enable programs" ON)
+option(CVD_EXAMPLES "Enable examples" ON)
+option(CVD_FAST_OBSCURE "Enable obscure fast algorithms" ON)
+
+if(CVD_SSE2)
+	set(CVD_SSE ON)
+endif()
+
+if(NOT MSVC)
+	if(CVD_SSE2)
+		add_compile_options(-msse2)
+	endif()
+
+	if(CVD_SSE)
+		add_compile_options(-mfpmath=sse)
```

Comment:
I think we just want -msse here: it'd whether or not the compiler defines the SSE builtins. The -mfpmath thing should just be done by normal CXXFLAGS I think if the user wants it. Is there an OK way to make SSE/SSE2 default to ON for x86 platforms, since there hasn't been an x86 CPU released without SSE2 in over a decade.